pkg/ingester/wal.go

```diff
@@ -241,6 +241,24 @@ func (w *walWrapper) performCheckpoint() (err error) {
 		os.RemoveAll(checkpointDirTemp)
 	}()
 
+	// Count number of series - we'll use this to rate limit checkpoints.
+	numSeries := 0
+	for _, state := range w.getUserStates() {
+		numSeries += state.fpToSeries.length()
+	}
+	if numSeries == 0 {
+		return nil
+	}
+
+	perSeriesDuration := w.cfg.CheckpointDuration / time.Duration(numSeries)
+	ticker := time.NewTicker(perSeriesDuration)
+
+	if immediate {
+		ticker.Stop()
```

Comment:
What's the point of creating a `ticker` we immediately `Stop()`?

Reply:
I guess I was in a hurry :) I have modified it now to only initialize when needed